File: src/ros2_bridge/mappers/sensor_msgs/illuminance.rs
//! Typed mapper for `sensor_msgs/msg/Illuminance`.

use crate::ros2_bridge::mapper::TypedTopicMapper;

pub(crate) fn illuminance_to_bus(
    msg: ros_env::sensor_msgs::msg::Illuminance,
) -> crate::sensor_msgs::msg::v1::Illuminance {
    crate::sensor_msgs::msg::v1::Illuminance {
        header: Some(crate::ros2_bridge::mappers::std_msgs::header::header_to_bus(msg.header)),
        illuminance: msg.illuminance,
        variance: msg.variance,
    }
}

pub(crate) fn illuminance_to_ros(
    bus: crate::sensor_msgs::msg::v1::Illuminance,
) -> ros_env::sensor_msgs::msg::Illuminance {
    ros_env::sensor_msgs::msg::Illuminance {
        header: crate::ros2_bridge::mappers::std_msgs::header::header_to_ros(
            bus.header.unwrap_or_default(),
        ),
        illuminance: bus.illuminance,
        variance: bus.variance,
    }
}

#[derive(Clone, Copy, Debug, Default)]
pub struct SensorMsgsIlluminanceMapper;

impl TypedTopicMapper for SensorMsgsIlluminanceMapper {
    type Ros = ros_env::sensor_msgs::msg::Illuminance;
    type Bus = crate::sensor_msgs::msg::v1::Illuminance;

    fn ros_to_bus(&self, msg: Self::Ros) -> crate::errors::Result<Self::Bus> {
        Ok(illuminance_to_bus(msg))
    }

    fn bus_to_ros(&self, msg: Self::Bus) -> crate::errors::Result<Self::Ros> {
        Ok(illuminance_to_ros(msg))
    }
}

File: bindings/python/robot_bus/ros2_bridge/mappers/sensor_msgs/illuminance.py
"""Generated mapper for `sensor_msgs/msg/Illuminance`."""

from __future__ import annotations

from robot_bus.ros2_bridge.mappers import _convert
from robot_bus.ros2_bridge.mappers.std_msgs.header import header_to_bus, header_to_ros

_BusMsg = None


def _bus_cls():
    global _BusMsg
    if _BusMsg is None:
        from robot_bus.sensor_msgs.msg.v1 import Illuminance as BusMsg

        _BusMsg = BusMsg
    return _BusMsg


def illuminance_to_bus(msg):
    BusMsg = _bus_cls()
    bus = BusMsg()
    bus.header.CopyFrom(header_to_bus(msg.header))
    bus.illuminance = msg.illuminance
    bus.variance = msg.variance
    return bus


def illuminance_to_ros(bus):
    from sensor_msgs.msg import Illuminance as RosMsg

    out = RosMsg()
    out.header = header_to_ros(bus.header)
    out.illuminance = bus.illuminance
    out.variance = bus.variance
    return out


class SensorMsgsIlluminanceMapper:
    _ros_type = None

    def ros_msg_type(self):
        cls = type(self)
        if cls._ros_type is None:
            from sensor_msgs.msg import Illuminance as RosMsg

            cls._ros_type = RosMsg
        return cls._ros_type

    def ros_to_bus(self, msg) -> bytes:
        return illuminance_to_bus(msg).SerializeToString()

    def bus_to_ros(self, payload: bytes):
        BusMsg = _bus_cls()
        bus = BusMsg()
        bus.ParseFromString(payload)
        return illuminance_to_ros(bus)
